### PR TITLE
python: Fix Bug in get_config_var()

### PR DESCRIPTION
**Pull request by citibeth**
New version 2.7.5
Keep clean environment when getting Python configuration; solves a bug.

### CODE REVIEW

**Comment by citibeth** [comment]
@adamjstewart Making recent change suggested by code review has made this not work.  How can we fix this?

**AttributeError: AttributeError: 'str' object has no attribute 'add_default_env'**


```
-        # Spack-installed Python works best without other env vars polluting it.
-        # https://askubuntu.com/questions/640010/fatal-python-error-py-initialize-unable-to-get-the-locale-encoding-syntaxe?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa
-        return self.command('-c', cmd, output=str, env={}).strip()
-
+        # Spack-installed Python works best
+        # without other env vars polluting it.
+        # https://askubuntu.com/questions/640010
+        pythoncmd = self.command('-c', cmd, output=str).strip()
+        pythoncmd.add_default_env('PYTHONPATH', '')
+        return pythoncmd
+        
```

**Comment by citibeth** [comment]
@adamjstewart The code currently in this PR is wrong, and I'm not sure how to fix it.  Do you have ideas?

File: var/spack/repos/builtin/packages/python/package.py [hunk 1]
```diff
@@ -454,7 +454,12 @@ def get_config_var(self, key):
         cmd = 'from distutils.sysconfig import get_config_var; '
         cmd += self.print_string("get_config_var('{0}')".format(key))
 
-        return self.command('-c', cmd, output=str).strip()
+        # Spack-installed Python works best
+        # without other env vars polluting it.
+        # https://askubuntu.com/questions/640010
+        pythoncmd = self.command('-c', cmd, output=str).strip()
+        pythoncmd.add_default_env('PYTHONPATH', '')
+        return pythoncmd
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/python/package.py]: `pythoncmd` is a string, you need to add the environment to the `Executable`. Can you try:
```python
self.command.add_default_env('PYTHONPATH', '')
return self.command('-c', cmd, output=str).strip()
```

**Comment by citibeth** [comment]
@adamjstewart I'm uncomfortable merging this because it "unexpectedly" changes the state of `self.command`.  I don't know how or where, but this kind of thing could cause "strange, hard-to-track-down" bugs elsewhere.  Can we just go back to the original suggestion in this PR, which does not change state?
```
        # Spack-installed Python works best without other env vars polluting it.
        # https://askubuntu.com/questions/640010/fatal-python-error-py-initialize-unable-to-get-the-locale-encoding-syntaxe?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa
        return self.command('-c', cmd, output=str, env={}).strip()
```
I could be more surgical in removing just `PYTHONPATH` from `env`, rather than clearing the whole thing.  How do I clone the environment currently residing in `self.command`?

**Comment by citibeth** [comment]
@adamjstewart Can you please give your thoughts to my latest post to this thread?

**Comment by adamjstewart** [comment]
@citibeth Can you try this and see if it solves your problem?
```python
temp_env = os.environ.copy()
temp_env.pop('PYTHONPATH', None)
return self.command('-c', cmd, output=str, env=temp_env).strip()
```

**Comment by citibeth** [comment]
I think the original version of this PR was a band-aid.  If the Spack-built Python command fails for `get_config_var()` due to misplaced `PYTHONPATH`, wouldn't it fail in all cases?  So I just changed the PR to fix the `PYTHONPATH` at the source, inside the `spack.command()` property.  I have tested this.

**Comment by adamjstewart** [comment]
Hmm, how did you test this? Doesn't spack set `PYTHONPATH` during package installation so that packages can find their dependencies?

**Comment by citibeth** [comment]
I unfortunately did not share information required to reproduce this bug, and I cannot do so now.  I will close this PR until/unless I encounter it again.

**Pull request closed without merging**